Update to gha-puppet v3 & voxpupuli-test 9.x

ekohl opened the pull request.

ekohl committed:
Update to gha-puppet v3 & voxpupuli-test 9.x

ekohl commented:
Wut?
```
Error: Facter: Error while resolving custom fact fact='rh_certificate', resolution='<anonymous>': break from proc-closure
```

ekohl committed:
Use next instead of break when fact is missing

In Ruby you're supposed to use next to exit a block, otherwise you get
errors like:

    Error: Facter: Error while resolving custom fact fact='rh_certificate', resolution='<anonymous>': break from proc-closure

ekohl commented:
> `Error: Facter: Error while resolving custom fact fact='rh_certificate', resolution='<anonymous>': break from proc-closure`

You're supposed to use `next`.

Now for the real failure: I think this is because a bash script has replaced the Ruby package so checks are failing.

@ofedoren I think that module effectively becomes obsolete, correct?

ofedoren commented:
> 
> Now for the real failure: I think this is because a bash script has replaced the Ruby package so checks are failing.

Not sure how bash package would affect this... I mean, I tested with Ansible role and it worked. We probably need to adjust Puppet class somehow though?..
> 
> @ofedoren I think that module effectively becomes obsolete, correct?

You mean the entire `puppet_foreman_scap_client`? I wouldn't say so. We still need this to automatically install the package and generate the config file. Although, at some point, we'd need to explicitly install Bash version here instead of Ruby one.

ekohl committed:
Pin acceptance tests to 3.13

Since 3.14 the Ruby implementation is replaced by bash. For now this
module is incompatible with those newer versions.

evgeni commented:
My understanding is that the new bash implementation is backwards compatible to the Ruby one, it uses the same config file, same executable name, everything. So unless this changes, this module is still usable (and required) as before. It "just" needs to know the right package name to become idempotent (again).
